### ISSUE TITLE
Inconsistent Pivoting

### ISSUE THREAD

**Issue by gilhs** [metabase/metabase#4210]
- Your browser and the version: Chrome 55
- Your operating system: OS X 10.12
- Your databases: BigQuery
- Metabase version: 0.21

I have a dashboard with a bunch of cards - all of which are custom SQL. A couple of those cards have a table visualization, with pivot enabled. The dashboard also has a text filter which affects all the cards (account filter). 

Once I apply a filter value, the pivoted tables are refreshed. However, the structure of the pivot changes (columns and rows are switched).

Before Filter:
![image](https://cloud.githubusercontent.com/assets/22723398/22181353/8148a2be-e093-11e6-9bb7-5580e86a92f4.png)

After Filter:
![image](https://cloud.githubusercontent.com/assets/22723398/22181354/84907ea6-e093-11e6-93a4-2da16cf04be2.png)

I think the better solution (which might require a separate issue?) would be to enable control of the pivoted data (what goes where).

**Comment by salsakran** [comment]
Yup, that's a bug. Re your better solution suggestion, we're going to significantly change how pivot tables work in our new query builder design (which will carry over into SQL questions as well).

**Comment by salsakran** [comment]
I think the specific thing happening here is that we will use the lower cardinality dimension as the column headers (as it's more of a pain to scroll horizontally), and the filter is causing the number of days to be lower than the number of users.

**Comment by tlrobinson** [comment]
Related to (duplicate of?) #2618

**Comment by gilhs** [comment]
@salsakran Been a few of months since this was reported, and I'm still encountering issues where the lack of ability to control the directionality of the pivot is limiting from producing the visualizations I'm looking for.

Is there any updated ETA on when this might be resolved (some way or another)?

**Comment by extraymond** [comment]
Hi! Would like to know if there is any further effort to make this work.

Currently I'm using filter in dashboard so I can let user view side-by-side comparison table.
But since the orientation of the pivot table is not able to be mandated, sometimes the comparing tables will have different orientation. This makes comparing two tables in a dashboard not ideal since the column directions are now messed up.
